server/manager/src/main/java/org/apache/accumulo/manager/tableOps/split/FindSplits.java

```diff
@@ -73,15 +95,59 @@ public Repo<Manager> call(FateId fateId, Manager manager) throws Exception {
+        if (result.getStatus() == Status.REJECTED) {
+          log.debug("{} unsplittable metadata update for {} was rejected ", fateId,
+              result.getExtent());
+        }
+      };
+
+      try (var tabletsMutator = ample.conditionallyMutateTablets(resultConsumer)) {
+        // No split points were found, so we need to check if the tablet still
+        // needs to be split but is unsplittable, or if a split is not needed
+
+        // Case 1: If a split is needed then set the unsplittable marker as no split
+        // points could be found so that we don't keep trying again until the
+        // split metadata is changed
+        if (SplitUtils.needsSplit(manager.getContext(), tabletMetadata)) {
+          log.info("Tablet {} needs to split, but no split points could be found.",
+              tabletMetadata.getExtent());
+          var unSplittableMeta = computedUnsplittable
+              .orElseGet(() -> SplitUtils.toUnSplittable(manager.getContext(), tabletMetadata));
+
+          // With the current design we don't need to require the files to be the same
+          // for correctness as the TabletManagementIterator will detect the difference
+          // when computing the hash and retry a new split operation if there is not a match.
+          // But if we already know there's a change now, it would be more efficient to fail and
+          // retry the current fate op vs completing and having the iterator submit a new one.
+          log.debug("Setting unsplittable metadata on tablet {}. hashCode: {}",
+              tabletMetadata.getExtent(), unSplittableMeta);
+          var mutator = tabletsMutator.mutateTablet(extent).requireAbsentOperation()
+              .requireSame(tabletMetadata, FILES).setUnSplittable(unSplittableMeta);
+          mutator.submit(tm -> unSplittableMeta.equals(tm.getUnSplittable()));
+
+          // Case 2: If the unsplittable marker has already been previously set, but we do not need
+          // to split then clear the marker. This could happen in some scenarios such as
+          // a compaction that shrinks a previously unsplittable tablet below the threshold
+          // or if the threshold has been raised higher because the tablet management iterator
+          // will try and split any time the computed metadata changes.
+        } else if (tabletMetadata.getUnSplittable() != null) {
+          log.info("Tablet {} no longer needs to split, deleting unsplittable marker.",
+              tabletMetadata.getExtent());
+          var mutator = tabletsMutator.mutateTablet(extent).requireAbsentOperation()
+              .requireSame(tabletMetadata, FILES).deleteUnSplittable();
+          mutator.submit(tm -> tm.getUnSplittable() == null);
+          // Case 3: The table config and/or set of files changed since the tablet mgmt iterator
+          // examined this tablet.
+        } else {
+          log.info("Tablet {} no longer needs to split, ignoring it.", tabletMetadata.getExtent());
```

Comment:
I would lower the log level on this, unless there is a bug with splitting its not useful.


```suggestion
          log.debug("Tablet {} no longer needs to split, ignoring it.", tabletMetadata.getExtent());
```